api/catalog/api/serializers/audio_serializers.py

```diff
@@ -64,107 +45,94 @@ class AudioSearchRequestSerializer(
-        return value.lower()
 
-    @staticmethod
-    def validate_length(value):
-        _validate_enum("length", LENGTHS, value)
-        return value.lower()
+class AudioReportRequestSerializer(MediaReportRequestSerializer):
+    class Meta:
+        model = AudioReport
+        fields = MediaReportRequestSerializer.Meta.fields
+        read_only_fields = MediaReportRequestSerializer.Meta.read_only_fields
 
 
-class AudioSerializer(MediaSerializer):
-    """A single audio file. Used in search results."""
+########################
+# Response serializers #
+########################
 
-    fields_names = [
-        *MediaSerializer.fields_names,
-        "audio_set",
-        "genre",
-        "duration",
-        "bit_rate",
-        "sample_rate",
-        "alt_files",
-        "detail_url",
-        "related_url",
-        "category",
-    ]
-    """
-    Keep the fields names in sync with the actual fields below as this list is
-    used to generate Swagger documentation.
-    """
 
-    audio_set = AudioSetSerializer(
-        required=False,
-        help_text="Reference to set of which this track is a part.",
-        read_only=True,
-    )
+class AudioSetSerializer(serializers.ModelSerializer):
+    """An audio set, rendered as a part of the ``AudioSerializer`` output."""
 
-    genres = serializers.ListField(
-        child=serializers.CharField(),
-        required=False,
-        help_text="An array of audio genres such as "
-        "`rock`, `electronic` for `music` category, or "
-        "`politics`, `sport`, `education` for `podcast` category",
-    )
+    class Meta:
+        model = AudioSet
+        fields = [
+            "title",
+            "foreign_landing_url",
+            "creator",
+            "creator_url",
+            "url",
+            "filesize",
+            "filetype",
+        ]
 
-    duration = serializers.IntegerField(
-        required=False, help_text="The time length of the audio file in milliseconds."
-    )
-    bit_rate = serializers.IntegerField(
-        required=False, help_text="Number in bits per second, eg. 128000."
-    )
-    sample_rate = serializers.IntegerField(
-        required=False, help_text="Number in hertz, eg. 44100."
-    )
 
-    alt_files = serializers.JSONField(
-        required=False, help_text="JSON describing alternative files for this audio."
-    )
+AudioHyperlinksSerializer = get_hyperlinks_serializer("audio")
+
+
+class AudioSerializer(AudioHyperlinksSerializer, MediaSerializer):
+    """A single audio file. Used in search results."""
+
+    class Meta:
+        model = Audio
+        fields = sorted(  # keep this list ordered logically
+            [
+                *MediaSerializer.Meta.fields,
+                *AudioHyperlinksSerializer.field_names,
+                "genres",
+                "alt_files",
+                "audio_set",
+                "duration",
+                "bit_rate",
+                "sample_rate",
+                "waveform",
```

Comment:
What is the difference between `waveform` and `peaks`? `peaks` is the array of numbers, and `waveform`?

Reply:
Added explanatory comments in 3d68558.